src/parser/jobs/ast/modules/Draw.tsx

```diff
@@ -0,0 +1,180 @@
+import {t} from '@lingui/macro'
+import {Plural, Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+// import {getDataBy} from 'data'
+import ACTIONS from 'data/ACTIONS'
+import {CastEvent} from 'fflogs'
+// import STATUSES from 'data/STATUSES'
+import Module, {dependency} from 'parser/core/Module'
+import Checklist, {Requirement, TARGET, TieredRule} from 'parser/core/modules/Checklist'
+import Suggestions, {SEVERITY, Suggestion, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import Cooldowns from 'parser/jobs/mch/modules/Cooldowns'
+import React from 'react'
+import {PLAY} from './ArcanaGroups'
+
+// THINGS TO TRACK:
+// Whether they used draw prepull (check how soon they played and then drew again)
+// perhaps go for displaying 11 out of 10 plays used, since prepull draw can be optional?
+// Track them using Draw when they still have a minor arcana (oopsie) or a card in the spread
+
+const CARD_DURATION = 15000
+const SLEEVE_DRAW_PLAYS_GIVEN = 3
+const SLEEVE_DRAW_CD_REDUCTION = 27
+const DRAW_CD_REFUND_PER_SLEEVE_PLAY = 3000
+const DRAW_CD_REFUND_PER_REDRAW_DURING_SLEEVE = 3000
+
+const WARN_TARGET_MAX_PLAYS = 2
+const FAIL_TARGET_MAX_PLAYS = 3
+
+const SEVERITIES = {
+	CARD_HOLDING: {
+		15000: SEVERITY.MINOR,
+		45000: SEVERITY.MEDIUM,
+		60000: SEVERITY.MAJOR,
+	},
+	SLEEVE_DRAW_OVERWRITE: {
+		15000: SEVERITY.MINOR,
+		45000: SEVERITY.MEDIUM,
+		60000: SEVERITY.MAJOR,
+	},
+}
+
+export default class Draw extends Module {
+	static handle = 'draw'
+	static title = t('ast.draw.title')`Draw`
+
+	@dependency private cooldowns!: Cooldowns
+	@dependency private checklist!: Checklist
+	@dependency private suggestions!: Suggestions
+
+	_lastDrawTimestamp = 0
+	_draws = 0
+	_drawDrift = 0
+	_drawTotalDrift = 0
+
+	_plays = 0
+
+// tslint:disable-next-line: no-magic-numbers
+	_sleeveResets: 0 | 1 | 2 | 3 = 0
+	_sleeveUses = 0
+	_sleeveOverwriteTime = 0
+
+	protected init() {
+		this.addHook('cast', {abilityId: ACTIONS.DRAW.id, by: 'player'}, this._onDraw)
+		this.addHook('cast', {abilityId: ACTIONS.SLEEVE_DRAW.id, by: 'player'}, this._onSleeveDraw)
+		this.addHook('cast', {abilityId: [...PLAY], by: 'player'}, this._onPlay)
+		this.addHook('complete', this._onComplete)
+	}
+
+	private _onDraw(event: CastEvent) {
+		this._draws++
+
+		if (this._draws === 1) {
```

Comment:
I had a quick look, and it _looks_ (sorry if i missed one!) like this is only being used to check the first - if that's the case, _mild_ preference for `firstDraw` as a `boolean` rather than counting each time.